pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -386,6 +387,12 @@ def dump_checkpoint(self, weights_only: bool = False) -> dict:
                 checkpoint[datamodule.__class__.__qualname__] = datamodule_state_dict
 
         # on_save_checkpoint hooks
+        if not weights_only:
```

Comment:
I don't understand how this is backward compatible. If someone wants to return state until 1.8, they should be able to do so.

Reply:
Yes returning state in `on_save_checkpoint` is supported (and overrides `state_dict`) until 1.8 (see `test_resume_callback_state_all` as well as `_call_callbacks_on_save_checkpoint` docstring)